src/Microsoft.OData.Core/ODataAnnotatable.cs

```diff
@@ -38,7 +38,12 @@ public abstract class ODataAnnotatable
         /// <returns>The custom instance annotations.</returns>
         internal ICollection<ODataInstanceAnnotation> GetInstanceAnnotations()
         {
-            Debug.Assert(this.instanceAnnotations != null, "this.instanceAnnotations != null");
+            //Debug.Assert(this.instanceAnnotations != null, "this.instanceAnnotations != null");
+            if (this.instanceAnnotations == null)
+            {
+                this.instanceAnnotations = new List<ODataInstanceAnnotation>();

```

Comment:
To the best of my knowledge, Collections should be used if I want to roll my own collection type with different behavior (it's more extendable). And since the list does not have any virtual methods, it should be slightly faster (though that advantage might be diminished because it's exposed via the `ICollection<T>` interface.